### DISCUSSION TITLE
Is there a mechanism for importing csv tables into a `qmd` file?

### DISCUSSION THREAD

**Discussion by mpf** [quarto-dev/quarto-cli#8254] [Q&A] [answered]
### Description

Does there exist a mechanism in to include a `csv` file into a markdown table in a `qmd` file, eg,

```
{{< include table.csv >}}
```

It's often convenient to maintain a csv file as a spreadsheet and then import it into the qmd file as markdown.

**Comment by mcanouil** [comment]
What does "include/import a csv" means?

To "import", you can use R/Python/Observable/etc.
To allow users/readers to download the file, simply create a link to that file `[My CSV file for you](path/to/my.csv)`.

PS: please do not mix up things. The include shortcode has a purpose which is to "copy/paste" a Quarto document into another. Avoid using things that have a very well defined role in order to not create confusion.

**Reply by mpf** [reply]
The aim is to read the contents of a csv file, convert the contents into a markdown formatted table, and past the results into the current file. [Franklin.jl](https://franklinjl.org/syntax/markdown/#inserting_a_table) has such functionality.

**Reply by mcanouil** [reply, marked as answer]
Then use code, for example with R:

```r
knitr::kable(read.csv("my.csv"))
```

Use Julia if you want to.

Nearly all computing languages has what's needed to do that.

I am starting to get a bit confused: why not using computations to compute (read from CSV to write as markdown)?
This is the whole purpose of Quarto.
